sys/cheri/cheri_sysctl.c

```diff
@@ -84,3 +90,43 @@ u_int	security_cheri_bound_legacy_capabilities;
 SYSCTL_INT(_security_cheri, OID_AUTO, bound_legacy_capabilities,
     CTLFLAG_RWTUN, &security_cheri_bound_legacy_capabilities, 0,
     "Set bounds on userspace capabilities created by legacy ABIs.");
+
+static void
+measure_cloadtags_stride(void *dummy __unused)
+{
+	void * __capability *buf;
+	uint64_t tags;
+	u_int i;
+
+	/*
+	 * Malloc a buffer as allocating an aligned page on the stack
+	 * risks overflowing the stack.
+	 *
+	 * Note that the buffer must not be simply aligned on a
+	 * capability boundary but aligned on a stride of
+	 * capabilities.
+	 */
+	buf = malloc_aligned(sizeof(*buf) * 64, sizeof(*buf) * 64,
+	    M_TEMP, M_WAITOK | M_ZERO);
+
+#ifdef INVARIANTS
+	tags = cheri_loadtags(buf);
+
+	KASSERT(tags == 0, ("CLoadTags on a zeroed buffer returned %lu", tags));
+#endif
+
+	/* CLoadTags can't return more than 64 bits. */
+	for (i = 0; i < 64; i++)
+		buf[i] = userspace_root_cap;
```

Comment:
The original code from caprevoke zeroed the buffer before `free()` and @jrtc27 noted that we likely leak this cap in other places so I removed the `memset()`.  If we are really worried about it, I think I'd just make it use `zfree` instead of plain `free`.

Reply:
Uses `zfree()` now.